### PR TITLE
refactor: Unsynced local docs

### PR DESCRIPTION
**Pull request by zoepage**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,16 +16,16 @@ check if local DB has unsynced changes.
...
-api.getLocalChanges()
+api.unsyncesLocalDocs()
 .then(function(changes) {
   // changes is array of docs?
 })
 
-api.getLocalChanges('docid') // pass doc id
-api.getLocalChanges({_id: 'docid'}) // pass doc with _id property
-api.getLocalChanges(['docid', {_id: 'docid'}]) // array of IDs and objects
+api.unsyncesLocalDocs('docid') // pass doc id
```
Comment (HipsterBrown) [inline README.md]: Minor typo here I think. 

``` js
api.unsyncedLocalDocs('docid')
```

Comment (zoepage) [inline README.md]: Very good catch. Thanks :)

File: README.md [hunk 2]
```diff
@@ -10,16 +16,16 @@ check if local DB has unsynced changes.
 // Initialisation
 
 var db = new PouchDB('dbname')
-var api = db.hoodieLocalChanges({remote: 'http://example.com/mydb'})
+var api = db.unsyncesLocalDocs({remote: 'http://example.com/mydb'})
```
Comment (gr2m) [inline README.md]: there is no db.unsyncesLocalDocs() that returns an `api` instance, on which you'd call a method any more. There is only a `db.unsyncesLocalDocs` method now, that you can call like this:

``` js
db.unsyncedLocalDocs({
  remote: 'http://example.com/mydb',
  // optional
  keys: [id1, id2]
}).then(function (docs) {
  // docs is array of docs
})
```

**Comment by gr2m** [comment]
@zoepage let me know if my comment here doesn't make sense to you: https://github.com/hoodiehq/pouchdb-hoodie-unsynced-local-docs/pull/6#discussion_r34170201

We need a lot more refactoring because of this, the implementation & tests are still testing as if we'd do this: 

``` js
var db = new PouchDB('dbname')
var api = db.unsyncedLocalDocs({remote: 'http://example.com/mydb'})

api.unsyncedLocalDocs()
.then(function(changes) {
  // changes is array of docs?
})
```

But instead it's now

``` js
var db = new PouchDB('dbname')

db.unsyncedLocalDocs({remote: 'http://example.com/mydb'})
.then(function(docs) { ... })
```

File: README.md (hunk 2 above)
Comment (zoepage) [inline README.md]: But we still return an array, not a promise right?

**Comment by zoepage** [comment]
@gr2m I had to give it a thought. 
As it's an internal method now, it makes sense. Will refactor today.

**Comment by zoepage** [comment]
refactor done by c2f69c3 and 05fc61f
cc @gr2m

**Comment by zoepage** [comment]
Last comment on your PR @gr2m. If it's okay, I'll finish the README and we're good.

**Comment by gr2m** [comment]
merged via 76a7df6 :tada:

**Comment by zoepage** [comment]
we can close this, right?

**Comment by gr2m** [comment]
jup, sorry :)

**Pull request closed without merging**